src/Framework.Configurator/Handlers/CreatePrincipalHandler.cs

```diff
@@ -1,28 +1,31 @@
-﻿using System.Threading.Tasks;
+﻿using System.Threading;
+using System.Threading.Tasks;
 
-using Framework.Authorization.BLL;
+using Framework.Authorization.BLL.Core.Context;
 using Framework.Authorization.Domain;
 using Framework.Configurator.Interfaces;
-using Framework.DomainDriven;
-using Framework.DomainDriven.BLL.Security;
 using Framework.SecuritySystem;
 
 using Microsoft.AspNetCore.Http;
 
 namespace Framework.Configurator.Handlers;
 
-public class CreatePrincipalHandler : BaseWriteHandler, ICreatePrincipalHandler
-
+public record CreatePrincipalHandler(
+        IAuthorizationRepositoryFactory<Principal> AuthorizationRepositoryFactory,
+        IConfiguratorIntegrationEvents? ConfiguratorIntegrationEvents = null) : BaseWriteHandler, ICreatePrincipalHandler
```

Comment:
how about a default/empty implementation (null object pattern)?

Reply:
No thanks :)

Reply:
I mean use the non-nullable argument for service, and register built-in service - it even exists - [SampleConfiguratorIntegrationEvents.cs](https://github.com/Luxoft/BSSFramework/blob/64f88dfc01e45be554958b515cdac5a5ebb382d4/src/_SampleSystem/SampleSystem.WebApiCore/Services/SampleConfiguratorIntegrationEvents.cs)`

so we can remove the conditional check for executing the method
```
if (this.ConfiguratorIntegrationEvents != null)
{
    await this.ConfiguratorIntegrationEvents.PrincipalCreatedAsync(principal, cancellationToken);
}
```